pact/ledger/ledger.pact

```diff
@@ -383,6 +395,22 @@
       ))
   )
 
+  (defun update-uri:bool
+    ( id:string
+      new-uri:string
+    )
+    (with-capability (UPDATE-URI-CALL id new-uri)
+      (policy-manager.enforce-update-uri (get-token-info id) new-uri)
+    )
+    (with-capability (UPDATE-URI id new-uri)
+      (with-read tokens id
+        { 'uri := old-uri }
+        (update tokens id { 'uri: new-uri })
```

Comment:
Thinking out loud.. Should we enforce that the `new-uri` is different than `old-uri`?
If not, I think we can remove the `with-read` here.